cmd/oracled/cmd/register_oracle.go

```diff
@@ -64,22 +66,59 @@ func registerOracleCmd() *cobra.Command {
 			defer queryClient.Close()
 
 			// get oracle account from mnemonic.
-			_, err = panacea.NewOracleAccount(conf.OracleMnemonic, conf.OracleAccNum, conf.OracleAccIndex)
+			oracleAccount, err := panacea.NewOracleAccount(conf.OracleMnemonic, conf.OracleAccNum, conf.OracleAccIndex)
 			if err != nil {
 				return fmt.Errorf("failed to get oracle account from mnemonic: %w", err)
 			}
 
 			// generate node key and its remote report
-			_, nodePubKeyRemoteReport, err := generateSealedNodeKey(nodePrivKeyPath)
+			nodePubKey, nodePubKeyRemoteReport, err := generateSealedNodeKey(nodePrivKeyPath)
 			if err != nil {
 				return fmt.Errorf("failed to generate node key pair: %w", err)
 			}
 
 			report, _ := enclave.VerifyRemoteReport(nodePubKeyRemoteReport)
-			_ = hex.EncodeToString(report.UniqueID)
+			uniqueID := hex.EncodeToString(report.UniqueID)
 
 			// request register oracle Tx to Panacea
-			// TODO: add register-oracle Tx
+			oracleCommissionRateStr, err := cmd.Flags().GetString(flagOracleCommissionRate)
+			if err != nil {
+				return err
+			}
+
+			oracleCommissionRate, err := sdk.NewDecFromStr(oracleCommissionRateStr)
+			if err != nil {
+				return err
+			}
+
+			msgRegisterOracle := oracletypes.NewMsgRegisterOracle(uniqueID, oracleAccount.GetAddress(), nodePubKey, nodePubKeyRemoteReport, trustedBlockInfo.TrustedBlockHeight, trustedBlockInfo.TrustedBlockHash, flagOracleEndpoint, oracleCommissionRate)
```

Comment:
`flagOracleEndPoint` should also change.

Reply:
Did you mean change to `cmd.Flags.GetString(flagOracleEndpoint)`?

Reply:
Ah, yes. Thanks for your hard work.